### PR TITLE
Add filter hook for adding options to google client.

### PR DESCRIPTION
**Pull request by stephen-czetty**
## Description
Added a hook for setting GMail-specific client options.

## Testing procedure
I have run in production for some time.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (modification of the currently available functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My code has appropriate phpdoc comments with a description, `@since`, `@params` and `@return`.
- [ ] My code is tested for both new installs and for users that are upgrading from older versions.

### CODE REVIEW

**Comment by slaFFik** [comment]
There should be a dot `.` at the end of the sentence (code comment), other than that - looks good and it will be implemented in the next WP Mail SMTP 1.5.0 release.

File: src/Providers/Gmail/Auth.php [hunk 1]
```diff
@@ -93,6 +93,8 @@ public function get_client() {
 		// We request only the sending capability, as it's what we only need to do.
 		$client->setScopes( array( \Google_Service_Gmail::MAIL_GOOGLE_COM ) );
 		$client->setRedirectUri( self::get_plugin_auth_url() );
+		// Apply custom options to the client
```
Comment (slaFFik) [inline src/Providers/Gmail/Auth.php:96]: Please add a dot `.` here.
This is a WordPress PHP Code Style rule, that we follow in this project.

Thank you!

**Review by slaFFik — APPROVED** [review]

File: src/Providers/Gmail/Auth.php [hunk 2]
```diff
@@ -93,6 +93,8 @@ public function get_client() {
...
+		$client = apply_filters('wp_mail_google_client_custom_options', $client);
```
Comment (jaredatch) [inline src/Providers/Gmail/Auth.php:97]: Spacing tweaks.
`apply_filters( 'wp_mail_google_client_custom_options', $client );`

**Comment by slaFFik** [comment]
Will be part of the next WP Mail SMTP version.

**Pull request closed without merging**